crates/bitcoin-da/src/rpc.rs

```diff
@@ -93,27 +90,44 @@ impl BitcoinNode {
                 "params": params
             }))
             .send()
-            .await;
-
-        // sometimes requests to bitcoind are dropped without a reason
-        // so impl. recursive retry
-        // TODO: add max retries
-        if let Err(error) = response {
-            // TODO: maybe remove is_request() check?
-            if error.is_connect() || error.is_timeout() || error.is_request() {
-                tokio::time::sleep(tokio::time::Duration::from_millis(500)).await;
-                return self.call(method, params).await;
-            }
-            return Err(anyhow!(error));
-        }
+            .await?;
 
-        let response = response.unwrap().json::<Response<T>>().await?;
+        let response = response.json::<Response<T>>().await?;
 
         if let Some(error) = response.error {
-            return Err(anyhow!(error));
+            return Ok(Err(error));
         }
 
-        Ok(response.result.unwrap())
+        Ok(Ok(response.result.unwrap()))
+    }
+
+    // TODO: add max retries
+    async fn call<T: serde::de::DeserializeOwned>(
+        &self,
+        method: &str,
+        params: Vec<serde_json::Value>,
+    ) -> Result<T, anyhow::Error> {
+        let mut attempt = 1;
+        loop {
```

Comment:
still looping?

I thought this pr was going to remove retrying and we'd do it in an outer scope?

Reply:
It's still looping, but not recursing.